### PR TITLE
[WhatsApp Business] New Actions

### LINKED ISSUE #4720: [ACTIONS] WhatsApp Business
### App
`whatsapp_business`
https://developers.facebook.com/docs/whatsapp/cloud-api

### Actions
- Send a text message
- List message templates
- Send a template message

Here are some examples:
https://pipedream.com/new?h=tch_wNxf8d

### PR DESCRIPTION
**Pull request by andrewjschuang**
**Actions:**

- List Message Templates
- Send Text Message
- Send Text Using Template

**Notes:**

- Phone Number ID is optional
- Templates with variables have a complex format for each template type, so at first they are being filtered out
- Configuration and account setup is very confusing, I can help with that

Resolves #4720.

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/whatsapp_business/actions/send-text-using-template/send-text-using-template.mjs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import whatsapp from "../../whatsapp_business.app.mjs";
+
+export default {
+  key: "whatsapp_business-send-text-using-template",
+  name: "Send Text Using Template",
+  description: "Send a text message using a pre-defined template",
```
Comment (jcortes) [inline components/whatsapp_business/actions/send-text-using-template/send-text-using-template.mjs:6]:
```suggestion
  description: "Send a text message using a pre-defined template. [See the docs](https://developers.facebook.com/docs/whatsapp/cloud-api/reference/messages)",
```

**Comment by DilanAthukorala** [comment]
@dannyroosevelt I am not able to test this with my facebook account. The access token genrated through Pipedream doesn't have the required permission scope to perform the actions. Is there an account (business account ID) that I can use to test this?

image
image

**Comment by DilanAthukorala** [comment]
> @DilanAthukorala how did you connect your account without entering a business account ID?

@dannyroosevelt I created a business account and was able to connect from that Business Account ID but the generated access token (Through Pipedream) does not have the required permissions to perform the actions. However I am able to perform these actions directly on the Business Account without using Pipedream.

**Comment by dannyroosevelt** [comment]
> > @DilanAthukorala how did you connect your account without entering a business account ID?
> 
> @dannyroosevelt I created a business account and was able to connect from that Business Account ID but the generated access token (Through Pipedream) does not have the required permissions to perform the actions. However I am able to perform these actions directly from the Business Account without using Pipedream.

Okay thanks for confirming, that's helpful. It might be because our app on WhatsApp / Facebook is not yet approved. I'll work on getting it approved so that you can QA.

**Review by dannyroosevelt — APPROVED** [review]
Confirming we'll ship these so the WhatsApp team can approve our app, and our QA team will QA and validate after that.

**Pull request merged**